library/modules/Military.cpp

```diff
@@ -0,0 +1,296 @@
+            order->min_count = squad_size;
+            //assumed
+            order->positions.resize(squad_size);
+
+            df::squad_order* s_order = df::allocate<df::squad_order_trainst>();
+
+            s_order->year = *df::global::cur_year;
+            s_order->year_tick = *df::global::cur_year_tick;
+
+            order->order = s_order;
+
+            asched[month].orders.push_back(order);
+            //wear uniform while training
+            asched[month].uniform_mode = 0;
+        };
+
+        //I thought this was a terrible hack, but its literally how dwarf fortress does it 1:1
+        //Off duty: No orders, Sleep/room at will. Equip/orders only
+        if (routine->name == "Off duty")
+        {
+            for (int i=0; i < 12; i++)
+            {
+                asched[i].sleep_mode = 0;
+                asched[i].uniform_mode = 1;
+            }
+        }
+        //Staggered Training: Training orders at 3 4 5, 9 10 11, sleep/room at will. Equip/orders only, except train months which are equip/always
+        //always seen the training indices 0 1 2 6 7 8, so its unclear. Check if squad id matters
+        else if (routine->name == "Staggered training")
+        {
+            //this is semi randomised for different squads
+            //appears to be something like squad.id & 1, it isn't smart
+            //if you alternate squad creation, its 'correctly' staggered
+            //but it'll also happily not stagger them if you eg delete a squad and make another
+            std::array<int, 6> indices;
+
+            if ((*df::global::squad_next_id) & 1)
+            {
+                indices = {3, 4, 5, 9, 10, 11};
+            }
+            else
+            {
+                indices = {0, 1, 2, 6, 7, 8};
+            }
+
+            for (int index : indices)
+            {
+                insert_training_order(index);
+                //still sleep in room at will even when training
+                asched[index].sleep_mode = 0;
+            }
+        }
+        //see above, but with all indices
+        else if (routine->name == "Constant training")
+        {
+            for (int i=0; i < 12; i++)
+            {
+                insert_training_order(i);
+                //still sleep in room at will even when training
+                asched[i].sleep_mode = 0;
+            }
+        }
+        else if (routine->name == "Ready")
+        {
+            for (int i=0; i < 12; i++)
+            {
+                asched[i].sleep_mode = 2;
+                asched[i].uniform_mode = 0;
+            }
+        }
+        else
+        {
+            for (int i=0; i < 12; i++)
+            {
+                asched[i].sleep_mode = 0;
+                asched[i].uniform_mode = 0;
+            }
+        }
+
+        result->schedule.push_back(reinterpret_cast<df::squad::T_schedule*>(asched));
+    }
+
+    //all we've done so far is leak memory if anything goes wrong
```

Comment:
where are we leaking? I don't see any `return` statements that come after `new` statements above

Reply:
I'm probably too used to code that can throw randomly, you're right in that there isn't currently a recoverable failure case where this can leak memory so I've removed this